src/components/date-picker/date-picker.ts

```diff
@@ -0,0 +1,771 @@
+
+  /** Determines whether the calendar hides its header.
+   * @attr hide-header
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'hide-header' })
+  public hideHeader = false;
+
+  /**
+   * Controls the visibility of the dates that do not belong to the current month.
+   * @attr hide-outside-days
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'hide-outside-days' })
+  public hideOutsideDays = false;
+
+  /** Gets/sets disabled dates. */
+  @property({ attribute: false })
+  public set disabledDates(dates: DateRangeDescriptor[]) {
+    this._disabledDates = dates;
+    this.setDateConstraints();
+    this.updateValidity();
+  }
+
+  public get disabledDates() {
+    return this._disabledDates as DateRangeDescriptor[];
+  }
+
+  /** Gets/sets special dates. */
+  @property({ attribute: false })
+  public specialDates!: DateRangeDescriptor[];
+
+  /**
+   * Whether the control will have outlined appearance.
+   * @attr
+   */
+  @property({ reflect: true, type: Boolean })
+  public outlined = false;
+
+  /**
+   * The placeholder attribute of the control.
+   * @attr
+   */
+  @property()
+  public placeholder!: string;
+
+  /**
+   * The number of months displayed in the calendar.
+   * @attr visible-months
+   */
+  @property({ type: Number, attribute: 'visible-months' })
+  public visibleMonths = 1;
+
+  /**
+   * Whether to show the number of the week in the calendar.
+   * @attr show-week-numbers
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'show-week-numbers' })
+  public showWeekNumbers = false;
+
+  /**
+   * Format to display the value in when not editing.
+   * Defaults to the input format if not set.
+   * @attr display-format
+   */
+  @property({ attribute: 'display-format' })
+  public set displayFormat(value: string) {
+    this._displayFormat = value;
+  }
+
+  public get displayFormat(): string {
+    return this._displayFormat ?? this.inputFormat;
+  }
+
+  /**
+   * The date format to apply on the input.
+   * Defaults to the current locale Intl.DateTimeFormat
+   * @attr input-format
+   */
+  @property({ attribute: 'input-format' })
+  public set inputFormat(value: string) {
+    this._inputFormat = value;
+  }
+
+  public get inputFormat(): string {
+    return this._inputFormat ?? this._input?.inputFormat;
+  }
+
+  /**
+   * The locale settings used to display the value.
+   * @attr
+   */
+  @property()
+  public locale = 'en';
+
+  /** The prompt symbol to use for unfilled parts of the mask.
+   *  @attr
+   */
+  @property()
+  public prompt = '_';
+
+  /** The resource strings of the calendar. */
+  @property({ attribute: false })
+  public resourceStrings: IgcCalendarResourceStrings =
+    IgcCalendarResourceStringEN;
+
+  @watch('open')
+  protected openChange() {
+    this._rootClickController.update();
+  }
+
+  /** Sets the start day of the week for the calendar. */
+  @property({ attribute: 'week-start' })
+  public weekStart:
```

Comment:
This is very analyzer-specific and can be addressed later/separately:
This is the same property as in the calendar re-exposed and if it's the same we might we as well create a shared type alias for both. Right now, those union types are a bit anonymous and for Blazor produce separate component-prefixed enumerations for the same thing, which is a bit redundant. Pretty sure there's a similar issue with the `setSelectionRange` direction and there are likely other as well